backend/src/components/auth.js

```diff
@@ -8,26 +8,39 @@ const qs = require('querystring');
 const utils = require('./utils');
 const HttpStatus = require('http-status-codes');
 const safeStringify = require('fast-safe-stringify');
-const {ApiError} = require('./error');
-const {pick} = require('lodash');
+const { ApiError } = require('./error');
+const { pick } = require('lodash');
 
 const auth = {
   // Check if JWT Access Token has expired
-  isTokenExpired(token) {
+  isTokenExpired(token, context = 'NA') {
     const now = Date.now().valueOf() / 1000;
     const payload = jsonwebtoken.decode(token);
+    log.info('isTokenExpired: context: ', context);
+    if (payload['exp']) {
+      // log.info('Token: JWT', token)
+      // log.info('isTokenExpired: exp:', payload['exp']);
+      // log.info('isTokenExpired: now: ', now + 30);
+      // log.info('isTokenExpired: diff: ', payload['exp'] - (now + 30));
+      const expDate = new Date((payload['exp'] * 1000));
+      log.info(`isTokenExpired: exp: ${expDate}, now: ${new Date()}`);
+    }
 
-    return (!!payload['exp'] && payload['exp'] < (now + 30)); // Add 30 seconds to make sure , edge case is avoided and token is refreshed.
+    return !!payload['exp'] && payload['exp'] < now + 30; // Add 30 seconds to make sure , edge case is avoided and token is refreshed.
   },
 
   // Check if JWT Refresh Token has expired
-  isRenewable(token) {
+  isRenewable(token, context = 'NA') {
     const now = Date.now().valueOf() / 1000;
     const payload = jsonwebtoken.decode(token);
-
+    if (!payload.iat || typeof payload.iat === 'undefined') {
+      log.warn('isRenewable', '| The token has no IAT possibly wrong token', payload);
```

Comment:
you should be able to just call `payload.iat === undefined`, unless its returning a string of undefined

Reply:
Try to address the null value. Here value could be 0.

Reply:
0 is valid in this situation? 

!payload.iat will catch 0, null, undefined, and false. 

Is this what you're trying to accomplish?
If (!payload.iat && payload.iat !== 0)

Reply:
0 is valid